src/ucp/core/ucp_ep.c

```diff
@@ -481,20 +552,20 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
     flags = UCP_PARAM_VALUE(EP, params, flags, FLAGS, 0);
     if (flags & UCP_EP_PARAMS_FLAGS_CLIENT_SERVER) {
         status = ucp_ep_create_to_sock_addr(worker, params, &ep);
-        if (status != UCS_OK) {
-            goto out;
-        }
-    } else {
+    } else if (params->field_mask & UCP_EP_PARAM_FIELD_EP_ADDR) {
+        status = ucp_ep_create_api_to_ep_addr(worker, params, &ep);
+        ucs_free(params->ep_addr);
```

Comment:
are we releasing the address even if the function failed?

Reply:
what is an other option to avoid memory leak? call "reject" by user in case of failure?

Reply:
if this is a transport level error then reject looks not completely working too...

Reply:
maybe rename "reject" to "abort"?

Reply:
"abort" sounds rough.. maybe add 3-rd function "ignore"?
really i don't see the reason why we can't release the handle here. if ucp can't create ep, the handle is useless.